### DISCUSSION TITLE
HTTP Endpoint Security

### DISCUSSION THREAD

**Discussion by ukepogluradr** [elsa-workflows/elsa-core#1470] [Q&A] [answered]
hi,

How do I secure HTTP Endpoints? Should I use any token based solution? Can I set own domain name for HTTP endpoints?

thanks in advance,
umit

**Comment by sfmskywalker** [comment, marked as answer]
To secure the API endpoints exposed by Elsa.Server.Api, you can do this from your Startup class:

```csharp
app.UseCors()
.UseRouting()
.UseAuthentication()
.UseAuthorization()
.UseHttpActivities()
.UseEndpoints(endpoints =>
{
    endpoints
       .MapControllers()
       .RequireAuthorization(); // <-- Add this to protect ALL controllers.
});
```

You can use any authentication service you like, such as JWT Bearer tokens, Open ID Connect or even your own custom implementations.

Elsa's HTTP Endpoint activities hook into ASP.NET's security pipeline, which means all you need to do to secure an HTTP Endpoint activity on the workflow is enable its `Authorize`  property.

Check out [this sample workflow](https://github.com/elsa-workflows/elsa-core/blob/master/src/samples/aspnet/Elsa.Samples.HttpEndpointSecurity/Workflows/SecureHelloWorkflow.cs) to see how that works.

> Can I set own domain name for HTTP endpoints?

HTTP Endpoint activities allow you to specify a relative path on your workflow server's domain. This means that you can't specify any domain that is not handled by your workflow server already. So the answer is yes, but you would do it on your hosting infrastructure.
